website/src/pages/DailyPage/DailyPage.tsx

```diff
@@ -22,41 +22,47 @@ import useApiCall from "../../utils/Hook";
 import ResponsiveChart from "./components/Chart";
 import DailySummary from "./components/DailySummary";
 import Hero from "./components/Hero";
+import { MacroData, MacroDataValue } from "@/types";
 
 import { secondToMicrosecond } from "../../utils/Utils";
 
+interface DailySummarydata {
+  name: string;
+  data : { total_qps: MacroDataValue }[];
+}
+
 export default function DailyPage() {
   const urlParams = new URLSearchParams(window.location.search);
-  const [benchmarkType, setBenchmarktype] = useState(
-    urlParams.get("type") == null ? "OLTP" : urlParams.get("type")
+  const [benchmarkType, setBenchmarktype] = useState<string>(
+    urlParams.get("type") ?? "OLTP"
   );
 
   const {
     data: dataDailySummary,
     isLoading: isLoadingDailySummary,
     error: errorDailySummary,
-  } = useApiCall(`${import.meta.env.VITE_API_URL}daily/summary`);
+  } = useApiCall<DailySummarydata>(`${import.meta.env.VITE_API_URL}daily/summary`);
 
   const {
     data: dataDaily,
     error: dailyError,
     textLoading: dailyTextLoading,
-  } = useApiCall(`${import.meta.env.VITE_API_URL}daily?type=${benchmarkType}`);
+  } = useApiCall<MacroData>(`${import.meta.env.VITE_API_URL}daily?type=${benchmarkType}`);
 
   const navigate = useNavigate();
 
   useEffect(() => {
     navigate(`?type=${benchmarkType}`);
   }, [benchmarkType]);
 
-  const TPSData = [
+  const TPSData: { id: string; data: { x: string; y: number }[] }[] = [
```

Comment:
I am thinking we should extract `{ x: string; y: number }` as a type, we are reusing this in a bunch of places.